rest/client/client.go

```diff
@@ -129,6 +138,55 @@ func (c *Client) newRequest(ctx context.Context, params Params, response interfa
 	return req, nil
 }
 
+func (c *Client) validateParams(params interface{}) error {
+	if err := c.validate.Struct(params); err != nil {
+		return fmt.Errorf("invalid request params: %w", err)
+	}
+	return nil
+}
+
+func (c *Client) encodePath(params interface{}) (map[string]string, error) {
+	val, err := c.pathEncoder.Encode(&params)
+	if err != nil {
+		return nil, fmt.Errorf("error encoding path params: %w", err)
+	}
+
+	pathParams := map[string]string{}
+	for k, v := range val {
+		pathParams[k] = v[0] // only accept the first one for a given key
+	}
+	return pathParams, nil
+}
+
+func (c *Client) encodePathString(uri string, params interface{}) (string, error) {
```

Comment:
this is all getting a bit messy since `newRequest` and `EncodeParams` do things differently. kinda wondering if we should just stop using resty to encode path and query params and do it ourselves since we already have to for iter methods